### PR TITLE
Polish modals

### PR DESCRIPTION
**Pull request by emmaknor**
- add icons to modals
- style modals
- add weight modal
- add weight API route
- modify dailyTracker test

### CODE REVIEW

File: _tests_/front-end/components/InsertModals.test.js [hunk 1]
```diff
@@ -1,10 +1,36 @@
-import {render} from '@testing-library/react';
+import {render, screen} from '@testing-library/react';
+import {GlobalStateProvider} from 'globalState.js';
 import '@testing-library/jest-dom';
 import * as React from 'react';
 import InsertModals from 'InsertModals.jsx';
 
-xdescribe('', () => {
-  it('', () => {
```
Comment (Ciaddie) [inline _tests_/front-end/components/InsertModals.test.js:7]: What are these lines doing?

**Review by Ciaddie — APPROVED** [review]

File: _tests_/front-end/components/InsertModals.test.js [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+describe('calorie, water, and weight modals', () => {
+  const renderModal = () => {
+    return render(<InsertModals />, {wrapper: GlobalStateProvider});
+  };
+
+
+  it('should render submit button', () => {
+    const {queryByText} = renderModal();
+    const {debug} = renderModal();
+    const btn = screen.queryByText('Record it!');
```
Comment (Scott-Guinn) [inline _tests_/front-end/components/InsertModals.test.js:20]: why do you need this 'screen' thing? I saw them doing this online too but I found that queryByText functions works on its own too? Do you know? 

(this is just a curiousity not a suggestion for change)

File: _tests_/front-end/components/InsertModals.test.js [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
+    console.log('btn!!! ', btn);
+    debug();
+    expect(btn).toBeTruthy();
   });
 });
+
+*/
+
+/*
+tests for modals:
```
Comment (Scott-Guinn) [inline _tests_/front-end/components/InsertModals.test.js:30]: ooh i like that you listed the tests here. Might steal.

**Review by mmoren01 — APPROVED** [review]

File: _tests_/front-end/components/dailyTracker.test.js [hunk 4]
```diff
@@ -1,25 +1,23 @@
-import {render, fireEvent, cleanup} from '@testing-library/react';
+import {render, fireEvent, screen} from '@testing-library/react';
```
Comment (Scott-Guinn) [inline _tests_/front-end/components/dailyTracker.test.js:1]: fireEvent sounds way more badass than userEvent

**Review by taylorsmart — COMMENTED** [review]
Approved

File: _tests_/front-end/components/InsertModals.test.js (hunk 2 above)
Comment (emmaknor) [reply _tests_/front-end/components/InsertModals.test.js:20]: I believe the benefit of the screen is that you have the ability to look within the entire document body. Does queryByText achieve the same thing?

File: _tests_/front-end/components/dailyTracker.test.js (hunk 4 above)
Comment (emmaknor) [reply _tests_/front-end/components/dailyTracker.test.js:1]: hahaha I agree

**Pull request merged**